asllib/Typing.ml

```diff
@@ -2917,6 +2903,77 @@ module Annotate (C : ANNOTATE_CONFIG) : S = struct
     |: TypingRule.AnnotateFuncSig
   (* End *)
 
+  module ControlFlow : sig
+    val check_stmt_interrupts : identifier -> env -> stmt_desc annotated -> prop
+    (** [check_stmt_interrupts name env body] checks that the function named
+        [name] with the statement body [body] returns a value or throws an
+        exception. *)
+  end = struct
+    (** Possible Control-Flow actions of a statement. *)
+    type t =
+      | Nothing  (** Continuing simply to the next statement. *)
```

Comment:
[Naming] In terms of a dataflow analysis, I think this is a may analysis with `Nothing` as the top element of the lattice. Essentially, I think it means "considering all the control flow paths of a given statement, there is at least one path that does not return a value or throws an exception."
`Nothing` doesn't convey any information and the comment I think does not reflect the property above.
How about something like `Top` or `MaybeErroneous`?

Reply:
I think this is okay.
I the lattice I am thinking of is the powerset of the kinds of runtime configurations that the semantics of a statement may result with: Returning/Continuing/Normal/Throwing, with union as the join operator.
This is slightly precise than needed, so I think you're using a further abstraction of that lattice where `Nothing` corresponds to the entire set of configurations (top), Interrupt corresponds to any non-empty subset of Returning/Throwing, and `Unreachable` corresponds to the empty set (bottom).